### PR TITLE
Restructure project

### PR DESCRIPTION
**Pull request by bregman-arie**
This includes the following changes:
* deleted select_age class. select_age is really only one
  function that can be included in select_related
* select_related renamed to resources which relfect better
  what the class does
* janitor main is shorter now. Added couple of functions
  which makes the code more readable and shorter
* more documentation
* improved some code blocks

### CODE REVIEW

File: shade_janitor/resources.py [hunk 1]
```diff
@@ -54,43 +48,36 @@ def _add_floatingip(self, fip):
...
 
-    def check_instance_blacklisted(self, instance):
-        skip_it = False
-        for entry in ['jenkins', 'slave', 'mirror']:
-            if entry in instance.name:
-                skip_it = True
-                break
-        return skip_it
+    def is_blacklisted(self, instance):
+        return instance.name in self.BLACKLIST
```
Comment (yazug) [inline shade_janitor/resources.py]: This logic change breaks the searching logic as instance.name in self.BLACKLIST doesn't check substrings

old logic would check for blacklisted instances

jenkins-foo
mirror-local-bar

Which is intended to be skipped

File: shade_janitor/resources.py [hunk 2]
```diff
@@ -54,43 +48,36 @@ def _add_floatingip(self, fip):
...
+                self._add_instance(instance)
 
-    def select_instances_name_substring(self, search_substring):
-        """
-        will select related resources based on provided substring
+    def select_instances_name_substring(self, search_str):
+        """Select related resources based on provided substring.
 
-        Excludes blacklisted instances
+        Excludes blacklisted instancesl.
```
Comment (yazug) [inline shade_janitor/resources.py]: typo "instances1."

File: shade_janitor/janitor.py [hunk 3]
```diff
@@ -30,17 +58,14 @@
...
-    # Cloud configs are read with os-client-config
-    cloud = shade.openstack_cloud(cloud=args.cloud)
+    cloud = initialize_cloud(args.cloud)
 
-    resources = SelectRelatedResources(cloud)
+    resources = Resources(cloud)
     cleanup = {}
 
     if args.old_instances:
-        age_resources = SelectAgeRelatedResources(cloud)
```
Comment (yazug) [inline shade_janitor/janitor.py]: I think this logic may end up cleaning up all old instances in one pass, which was not the intent here.  used 2 different Selectors to identify old instances and then cleanup the oldest one first.  I don't see necessary changes to drop everything from resources to make this work and maintainable.

File: shade_janitor/resources.py (hunk 1 above)
Comment (bregman-arie) [inline shade_janitor/resources.py]: This logic also checks for blacklisted instances
self.BLACKLIST is ['jenkins', 'slave', 'mirror']

Also, there is no substring in removed code/logic

File: shade_janitor/resources.py (hunk 2 above)
Comment (bregman-arie) [inline shade_janitor/resources.py]: Thanks

**Comment by bregman-arie** [comment]
We can close this pull request.
I'll break it into smaller parts.

**Pull request closed without merging**